Error for idp token expire even if IdpSignKeyRefreshEnabled enabled

Hi, I have netbird selfhosted that use idp google, initially after some days always make impossible login until reboot.
After found "IdpSignKeyRefreshEnabled" I enabled it and for long time I not had issue.
Today same issue appaired (at least seems), here latest logs before reboot (after reboot return to work correctly):
```
infrastructure_files-management-1  | 2023-12-12T09:12:27Z ERRO management/server/http/util/util.go:80: got a handler error: token invalid
infrastructure_files-management-1  | 2023-12-12T09:12:27Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 4056351373: GET /api/users status 401
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/jwtclaims/jwtValidator.go:158: error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/middleware/auth_middleware.go:65: Error when validating JWT claims: Error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/util/util.go:80: got a handler error: token invalid
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 1058967340: GET /api/users status 401
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/jwtclaims/jwtValidator.go:158: error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/middleware/auth_middleware.go:65: Error when validating JWT claims: Error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/util/util.go:80: got a handler error: token invalid
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 3694887989: GET /api/users status 401
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/jwtclaims/jwtValidator.go:158: error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/middleware/auth_middleware.go:65: Error when validating JWT claims: Error parsing token: unable to find appropriate key
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/http/util/util.go:80: got a handler error: token invalid
infrastructure_files-management-1  | 2023-12-12T09:12:28Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 3769408553: GET /api/groups status 401
```
I couldn't keep it not working for more in-depth debugging because we needed to use it, let me know if there is any more data I can provide even if the problem is no longer occurring.

--- a/management/cmd/management.go
+++ b/management/cmd/management.go
@@ -83,7 +83,10 @@ var (
 			if err != nil {
 				return fmt.Errorf("failed reading provided config file: %s: %v", mgmtConfig, err)
 			}
-			config.HttpConfig.IdpSignKeyRefreshEnabled = idpSignKeyRefreshEnabled
+
+			if cmd.Flag(idpSignKeyRefreshEnabledFlagName).Changed {
+				config.HttpConfig.IdpSignKeyRefreshEnabled = idpSignKeyRefreshEnabled
+			}
 
 			tlsEnabled := false
 			if mgmtLetsencryptDomain != "" || (config.HttpConfig.CertFile != "" && config.HttpConfig.CertKey != "") {

--- a/management/cmd/root.go
+++ b/management/cmd/root.go
@@ -12,7 +12,8 @@ import (
 
 const (
 	// ExitSetupFailed defines exit code
-	ExitSetupFailed = 1
+	ExitSetupFailed                  = 1
+	idpSignKeyRefreshEnabledFlagName = "idp-sign-key-refresh-enabled"
 )
 
 var (
@@ -62,7 +63,7 @@ func init() {
 	mgmtCmd.Flags().StringVar(&certKey, "cert-key", "", "Location of your SSL certificate private key. Can be used when you have an existing certificate and don't want a new certificate be generated automatically. If letsencrypt-domain is specified this property has no effect")
 	mgmtCmd.Flags().BoolVar(&disableMetrics, "disable-anonymous-metrics", false, "disables push of anonymous usage metrics to NetBird")
 	mgmtCmd.Flags().StringVar(&dnsDomain, "dns-domain", defaultSingleAccModeDomain, fmt.Sprintf("Domain used for peer resolution. This is appended to the peer's name, e.g. pi-server. %s. Max length is 192 characters to allow appending to a peer name with up to 63 characters.", defaultSingleAccModeDomain))
-	mgmtCmd.Flags().BoolVar(&idpSignKeyRefreshEnabled, "idp-sign-key-refresh-enabled", false, "Enable cache headers evaluation to determine signing key rotation period. This will refresh the signing key upon expiry.")
+	mgmtCmd.Flags().BoolVar(&idpSignKeyRefreshEnabled, idpSignKeyRefreshEnabledFlagName, false, "Enable cache headers evaluation to determine signing key rotation period. This will refresh the signing key upon expiry.")
 	mgmtCmd.Flags().BoolVar(&userDeleteFromIDPEnabled, "user-delete-from-idp", false, "Allows to delete user from IDP when user is deleted from account")
 	rootCmd.MarkFlagRequired("config") //nolint
 

--- a/management/server/jwtclaims/jwtValidator.go
+++ b/management/server/jwtclaims/jwtValidator.go
@@ -108,6 +108,8 @@ func NewJWTValidator(issuer string, audienceList []string, keysLocation string, 
 						refreshedKeys = keys
 					}
 
+					log.Debugf("keys refreshed, new UTC expiration time: %s", refreshedKeys.expiresInTime.UTC())
+
 					keys = refreshedKeys
 				}
 			}
@@ -179,7 +181,7 @@ func (m *JWTValidator) ValidateAndParse(token string) (*jwt.Token, error) {
 
 // stillValid returns true if the JSONWebKey still valid and have enough time to be used
 func (jwks *Jwks) stillValid() bool {
-	return jwks.expiresInTime.IsZero() && time.Now().Add(5*time.Second).Before(jwks.expiresInTime)
+	return !jwks.expiresInTime.IsZero() && time.Now().Add(5*time.Second).Before(jwks.expiresInTime)
 }
 
 func getPemKeys(keysLocation string) (*Jwks, error) {

